### ISSUE TITLE
Often seeing 99+ errors/warnings from lib.d.ts

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#3766]
More often these days (I have seen this before) my problems/warnings status bar shows 99+ and clicking on it shows all are coming from /Applications/Visual Studio Code - Alpha.app/Contents/Resources/app/extensions/typescript/server/typescript/lib/lib.d.ts

![image](https://cloud.githubusercontent.com/assets/900690/13567077/6337ffe0-e459-11e5-98e7-04f1983d07db.png)

In addition the console prints this:

[Extension Host] diagnostics for file:///Applications/Visual%20Studio%20Code%20-%20Alpha.app/Contents/Resources/app/extensions/typescript/server/typescript/lib/lib.d.ts will be capped to 250 (actually is 588)

**Comment by egamma** [comment]
@bpasero can you add how such an error now looks like. Makes me wonder whether the error comes from tslint or from ts. We now show the source of the error in the prefix, e.g., `[tslint] message`.

**Comment by bpasero** [comment]
@egamma 

![image](https://cloud.githubusercontent.com/assets/900690/13599765/26e24326-e525-11e5-99e6-b22b50e80fc1.png)

![image](https://cloud.githubusercontent.com/assets/900690/13599770/2e90471c-e525-11e5-95c8-e0784a588f49.png)

**Comment by dbaeumer** [comment]
@bpasero there is little I can do about that. The problems shows up if you configure the tsc/tsserver compiler with noLib:true and point to a different version of the lib (which we usually do for extensions). If you then open the lib version that ship with the tsserver in that context you get duplicate identifiers (which from a compiler perspective is actually correct). 

Do you have steps how to get there? IMO the shipped lib.d.ts should not be loaded in that context. So it is actually a tsserver bug. Would be could to give the TS team steps

**Comment by bpasero** [comment]
@dbaeumer I dont have steps other than I see this often per day during normal selfhost, I would be very surprised if I am the only one that hits this.

**Comment by bpasero** [comment]
Btw I have seen errors/warnings show errors from *.d.ts files ever since we have the errors/warnings indicator in the status bar, so it is not a recent thing.

**Comment by bpasero** [comment]
To be clear: I never open the d.ts files, so I am not sure how the errors are coming into the system at all. I suggest to add some instrumentation code to see who resolved it.

**Comment by felixfbecker** [comment]
I can confirm this. A "clear all errors" command would be helpful at least.

**Comment by dbaeumer** [comment]
@bpasero for the cases you reported I was able to track this down. The reason was a move of the tsconfig.json file from the src directory to its parent which included the server directory which has the tsserver in it a all the lib.*.d.ts files. (see your screen shot; all the errors point to the typescript extension server directory). I excluded the server directory from the tsconfig.json so you shouldn't see these errors again. I will close the issue since I could explain these errrors and fixed the tsconfig.json

@felixfbecker: I am against a clear errors action since something like that either points to setup problem or a bug in the tsserver. And this action would simply hide this.
